src/main/js/controls/Bar/helpers/legendHelpers.js

```diff
@@ -47,16 +47,55 @@ const clickHandler = (graphContext, control, config, canvasSVG) => (
             shownTargets.push(item.key);
         }
     };
+    if (config.shownTargets.length === 0) {
+        graphContext.contentKeys.forEach((i) => {
+            canvasSVG
+                .selectAll(`.${styles.taskBar}[aria-describedby="${i}"]`)
+                .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        });
+    }
     legendClickHandler(element);
     updateShownTarget(config.shownTargets, item);
-    canvasSVG
-        .selectAll(`g[aria-describedby="${item.key}"]`)
-        .attr("aria-hidden", isLegendSelected(d3.select(element)));
-    setSelectionIndicatorAttributes(
-        canvasSVG.selectAll(`.${styles.taskBarSelection}`),
-        false
-    );
-    window.requestAnimationFrame(onAnimationHandler(graphContext));
+    if (config.shownTargets.length >= 1) {
+        canvasSVG
+            .selectAll(`g[aria-describedby="${item.key}"]`)
+            .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        canvasSVG
+            .selectAll(`.${styles.taskBar}[aria-describedby="${item.key}"]`)
+            .attr("aria-hidden", isLegendSelected(d3.select(element)));
+        canvasSVG
+            .selectAll(
+                `.${styles.region}[aria-describedby="region_${item.key}"]`
+            )
+            .attr("aria-hidden", function () {
+                return !(d3.select(this).attr("aria-hidden") === "true");
+            });
+        setSelectionIndicatorAttributes(
+            canvasSVG.selectAll(`.${styles.taskBarSelection}`),
+            false
+        );
+        setSelectionIndicatorAttributes(
+            canvasSVG.selectAll(`.${styles.taskBarSelection}`),
+            false
+        );
```

Comment:
Is it intentional that we have this code duplicated?

Reply:
that is not intentional Andy and i removed duplicate code [here](https://github.com/cerner/carbon-graphs/pull/300/commits/595865c1ef97c1cd837971b727d58b1e7d89c5c0#diff-1ecc848586009dc5a2a7976d5f75492eL77-L80).